common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,13 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const record = { a: 1, b: 2 };
+const record1 = record.a; // This should not report an error
+const recordB = record.b; // This should not report an error
+console.log({ record1 });
+console.log({ recordB });
+type SomeObj = { [key: string]: string };
+const someObj: SomeObj = { a: "hello", b: "goodbye" };
+someObj.a.length; // This should report an error
```

Comment:
There are plenty more cases of interest.
The test code can have a lot of examples of problematic patterns and the alternatives that are okay. Comments for each of those should be added as well. Think about the information you'd want to see if one of these defects were raised.
For some of these cases, please see Teams discussions.

Reply:
@jason-ha are there any more that you would like that you dont see here?

Reply:
Yes, plenty more. What are the patterns that are okay?
In what cases is it okay to access `.a` from `DynamicType` (aka `SomeObj` in this thread)? Put those in the test file.

Reply:
Still need more. See prior comment.
When there is a problem case, what is/are the version(s) of the code that is/are okay?

Reply:
> Yes, plenty more. What are the patterns that are okay? In what cases is it okay to access `.a` from `DynamicType` (aka `SomeObj` in this thread)? Put those in the test file.

I was thinking accessing a property that is checked or the optional chaining operator are the only cases where it is ok to access a from a variable with type DynamicType? what about `someObj['a']` or someObj.a.length or Object.entries(someObjWithDynamicType), am I still missing more? I see a lot of tests for the original PR that introduced noUncheckedIndexedAccess;https://github.com/microsoft/TypeScript/pull/39560/files I dont know if most of them are useful to us

Reply:
https://teams.microsoft.com/l/message/19:8c0d3d2ff0374b19b5e46ad09dd5a613@thread.v2/1721926136798?context=%7B%22contextType%22%3A%22chat%22%7D
Has a lot of cases. I see more covered now, but there are still some missing. Go through the rules line by line and make sure there are positive and negative cases. Please also add comments in the test code more than should or should not be a defect - explain what is important / why defect or okay.

`Object.entries` is relevant if using a key it provides still triggers a defect. Doesn't hurt to have more cases tested.

In the end you should be able to run the test code with the noUncheckedIndexAccess and get defects at the same places (ignoring array cases where we are being explicitly different).

Here is one situation in three cases that I **think** should be covered:
```typescript
// 1
const aExpectingString: string = someObjWithDynamicType.a; // defect
// 2
const aImplicitType = someObjWithDynamicType.a; // no defect
aImplicitType.length; // defect
aImplicitType?.length; // no defect
// 3
const aExplicitStringOrUndefined: string | undefined = someObjWithDynamicType.a; // no defect
```

Reply:
There are more cases still. For example, item one from the referenced post, no defect if type already could be null or undefined per record type.